bin/check-solr-replication-lag.rb

```diff
@@ -12,10 +12,30 @@
 require 'date'
 
 class SolrCheckReplication < Sensu::Plugin::Check::CLI
-  option :core_url,
-         short: '-u URL',
-         long: '--url URL',
-         description: 'Solr Core Url',
+  option :host,
+         short: '-h HOST',
+         long: '--host HOST',
+         description: 'Solr Host to connect to',
+         required: true
+
+  option :port,
+         short: '-p PORT',
+         long: '--port PORT',
+         description: 'Solr Port to connect to',
+         proc: proc(&:to_i),
+         required: true
+
+  option :secure,
```

Comment:
rather than doing a conditional you can just pass it directly with what they pass via CLI. I would personally do something like this:
```ruby
option :protocol,
       long: '--protocol PROTOCOL',
       description: 'The connection protocol to use',
       in: %w[http https], # this controls the acceptable inputs
       default: 'http'
```